internal/descriptor/descriptor_v2.go

```diff
@@ -140,6 +140,18 @@ func PopV2(
 			}
 			fields := scalarFields2pX(r, descriptorsV2, unionOperation)
 			desc = V2{Compound, id, name, true, nil, fields}
+		case MultiRange:
+			name := r.PopString()
+			r.PopUint8() // schema_defined
+			ancestors := scalarFields2pX(r, descriptorsV2, false)
+			fields := []*FieldV2{{
+				Desc: descriptorsV2[r.PopUint16()],
+			}}
+			err := assertArrayDimensions(r)
+			if err != nil {
+				return V2{}, err
+			}
```

Comment:
I'm not sure that we need that here and for v1. If I got it right, multiranges descriptor looks similar to this

```c
struct MultiRangeTypeDescriptor {
    // Indicates that this is an
    // MultiRange Type descriptor.
    uint8   tag = 12;

    // Schema type ID.
    uuid    id;

    // Schema type name.
    string  name;

    // Whether the type is defined in the schema
    // or is ephemeral.
    bool    schema_defined;

    // Number of ancestor scalar types.
    uint16  ancestors_count;

    // Indexes of ancestor scalar type descriptors
    // in ancestor resolution order (C3).
    uint16  ancestors[ancestors_count];

    // Range type descriptor index.
    uint16  type;
};
```

`assertArrayDimensions` will attempt to read below the descriptor bytes and possibly corrupt the remainder of the `describe` message.